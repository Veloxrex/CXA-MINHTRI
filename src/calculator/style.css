@import url('https://fonts.googleapis.com/css?family=Space+Mono');


* {
    box-sizing: border-box;
}

.calculator input,
.calc-button   {
    font-family: 'Space Mono';
}


body {
    background: #F6F6F6;
    padding-bottom: 30px;
    padding-top: 30px;
}


.calculator   {
    max-width: 400px;
    margin: 0 auto;
    border: 1px solid #111;
    border-radius: 5px;
}

.calculator input   {
    background: none;
    border: none;
    box-shadow: none;
    outline: none;
    padding: 20px 10px;
    width: 100%;
    border-bottom: 2px solid #111;
    color: white;
    text-align: right;
    font-size: 40px;
    border-radius: 0;
    background: #433437;
}

.calc-button  {
    background: white;
    border: 1px solid rgba(17 , 17 , 17 , 0.5);
    padding: 20px;
    color: #111;
    font-size: 30px;
    cursor: pointer;
}

.calculator-buttons {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
}

.is-zero,
.is-clear   {
    grid-column: span 3;
}



.is-red {
    background: #F2274F !important;
    color: white;
    font-size: 30px;
}

