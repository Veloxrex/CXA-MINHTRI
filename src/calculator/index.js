import React, {useState} from "react"
import "./style.css"


const buttonArray = [
    {
        id: 'c',
        display: 'Clear'
    },
    {
        id: '/',
        display: "&#247;"
    },
    {
        id: 7,
        display: "7"
    },
    {
        id: 8,
        display: "8"
    },
    {
        id: 9,
        display: "9"
    },
    {
        id: '*',
        display: "&#215;"
    },
    {
        id: 4,
        display: "4"
    },
    {
        id: 5,
        display: "5"
    },
    {
        id: 6,
        display: "6"
    },
    {
        id: "-",
        display: "&#8722;"
    },
    {
        id: 1,
        display: "1"
    },
    {
        id: 2,
        display: "2"
    },
    {
        id: 3,
        display: "3"
    },
    {
        id: '+',
        display: "&#43;"
    },
    {
        id: 0,
        display: "0"
    },
    {
        id: '=',
        display: "&#61;"
    }

];
const genClassButton = (id) => {
    switch (id) {
        case 'c':
            return 'is-clear';
        case 0:
            return 'is-zero';
        case '=':
            return 'is-red';
        case '+':
            return 'is-red';
        case '-':
            return 'is-red';
        case '*':
            return 'is-red';
        case '/':
            return 'is-red';
        default:
            break;
    }
};

export const CalculatorComponent = () => {
    const [result, setResult] = useState('');
    const [equalStatus , setEqualStatus] = useState(false)
    const reset = () => {
        if (result !== '') {
            setEqualStatus(false)
            setResult('')
        }

    };

    const calculate = () => {
        try {

            // eslint-disable-next-line no-eval
            setResult((eval(result) || 0) + '');
            setEqualStatus(true)
        } catch (e) {
            setResult("error")
        }
    };

    const onClickKeyPad = (e) => {
        const button = e.target.id;
        if (button === "=") {
            calculate();
        } else if (button === "c") {
            reset()
        } else {
            if (!(["+", "-", "*", "/"].includes(button)) || result.slice(-1) !== button) {

                if(equalStatus){
                    reset();
                    setResult(button);
                }else{
                    setResult(result + button);
                }

            }

        }
    };

    return (
        <>
            {/*<span>{result}</span>*/}
            <div className="calculator">
                <input type="text" value={result} disabled/>
                <div className="calculator-buttons">
                    {buttonArray.map(item => {
                        const {id, display} = item;
                        return (
                            <button id={id} key={id} className={`calc-button ${genClassButton(id)}`}
                                    onClick={(e) => onClickKeyPad(e)}
                                    dangerouslySetInnerHTML={{__html: display}}/>
                        )
                    })}
                </div>

            </div>
        </>
    )
};



