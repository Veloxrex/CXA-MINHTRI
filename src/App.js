
import {CalculatorComponent} from "./calculator";
import React from "react";

function App() {
  return (
    <div className="App">
      {/*<CalculatorComponent/>*/}
    </div>
  );
}

export default App;
