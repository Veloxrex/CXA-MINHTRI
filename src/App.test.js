import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import {CalculatorComponent } from "./calculator";

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

it('renders calculator component without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<CalculatorComponent />, div);
});

// test('Link changes the class when hovered', () => {
//   const component = renderer.create(
//       <CalculatorComponent/>,
//   );
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
  //
  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
// });
