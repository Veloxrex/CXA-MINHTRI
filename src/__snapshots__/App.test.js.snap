// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="calculator"
>
  <input
    disabled={true}
    type="text"
    value=""
  />
  <div
    className="calculator-buttons"
  >
    <button
      className="calc-button is-clear"
      dangerouslySetInnerHTML={
        Object {
          "__html": "Clear",
        }
      }
      id="c"
      onClick={[Function]}
    />
    <button
      className="calc-button is-red"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#247;",
        }
      }
      id="/"
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "7",
        }
      }
      id={7}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "8",
        }
      }
      id={8}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "9",
        }
      }
      id={9}
      onClick={[Function]}
    />
    <button
      className="calc-button is-red"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#215;",
        }
      }
      id="*"
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "4",
        }
      }
      id={4}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "5",
        }
      }
      id={5}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "6",
        }
      }
      id={6}
      onClick={[Function]}
    />
    <button
      className="calc-button is-red"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#8722;",
        }
      }
      id="-"
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "1",
        }
      }
      id={1}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "2",
        }
      }
      id={2}
      onClick={[Function]}
    />
    <button
      className="calc-button undefined"
      dangerouslySetInnerHTML={
        Object {
          "__html": "3",
        }
      }
      id={3}
      onClick={[Function]}
    />
    <button
      className="calc-button is-red"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#43;",
        }
      }
      id="+"
      onClick={[Function]}
    />
    <button
      className="calc-button is-zero"
      dangerouslySetInnerHTML={
        Object {
          "__html": "0",
        }
      }
      id={0}
      onClick={[Function]}
    />
    <button
      className="calc-button is-red"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&#61;",
        }
      }
      id="="
      onClick={[Function]}
    />
  </div>
</div>
`;
